lib/hash.c

```diff
@@ -88,11 +88,11 @@ hash_alloc_intern (void *arg)
 
 #define hash_update_ssq(hz, old, new) \
   do { \
-    long double res; \
-    res = powl(old, 2.0); \
-    hz->stats.ssq -= (uint64_t) res;\
-    res = powl(new, 2.0); \
-    hz->stats.ssq += (uint64_t) res; \
+    uint32_t res; \
+    res = old * old; \
+    hz->stats.ssq -= res; \
+    res = new * new; \
+    hz->stats.ssq += res; \
```

Comment:
This needs to be a signle `atomic_fetch_add_explicit(&hz->stats.ssq, new - old, memory_order_explicit)`